starknet_devnet/server.py

```diff
@@ -204,19 +210,54 @@ def validate_load_messaging_contract(request_dict: dict):
         abort(Response(error_message, 400))
     return network_url
 
+@app.route("/dump", methods=["POST"])
+def dump():
+    """Dumps the starknet_wrapper"""
+
+    request_dict = request.json or {}
+    dump_path = request_dict.get("path") or dumper.dump_path
+    if not dump_path:
+        abort(Response("No path provided", 400))
+
+    dumper.dump(dump_path)
+    return Response(status=200)
+
+def dump_on_exit(_signum, _frame):
+    """Dumps on exit."""
+    dumper.dump(dumper.dump_path)
+    sys.exit(0)
+
 starknet_wrapper = StarknetWrapper()
+dumper = Dumper(starknet_wrapper)
```

Comment:
Why are this variables declared here and not somewhere at the top of the file?

Reply:
Back when the `main` function was mostly an `app` runner, the logic of instantiating a StarknetWrapper and parsing the args was done outside of it, so I guess this is a historic remnant. I could move them up.